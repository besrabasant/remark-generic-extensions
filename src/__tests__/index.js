import test from "ava"
import remark from "remark"
import html from "remark-html"

import genericExtensions from "../"

const transformToHtml = (input, options) => {
  const { contents } = remark()
    .use(genericExtensions, options)
    .use(html)
    .processSync(input)
  return contents
}

test(
  "should add an extension",
  t => t.snapshot(transformToHtml("!icon"))
)

test(
  "should add an extension with an id",
  t => t.snapshot(transformToHtml("!icon{ #my-id }"))
)

test(
  "should add an extension with a class",
  t => t.snapshot(transformToHtml("!icon{ .my-class }"))
)

test(
  "should add an extension with an attribute",
  t => t.snapshot(transformToHtml("!icon{ attr=my-attr }"))
)

test(
  "should add an extension with a quoted attribute",
  t => t.snapshot(transformToHtml(`!icon{ attr="my attribute" }`))
)

test(
  "should add an extension with a lone attribute",
  t => t.snapshot(transformToHtml(`!icon{ attr }`))
)

test(
  "should add an extension with a lone attribute with a default value",
  t => t.snapshot(
    transformToHtml(
      `!icon{ attr }`,
      {
        elements: {
          icon: {
            attributeDefaultValues: {
              attr: "default",
            }
          }
        }
      }
    )
  )
)

test(
  "should add an extension with a mapped content",
  t => t.snapshot(
    transformToHtml(
      "!icon[my-tooltip]",
      {
        elements: {
          icon: {
            attributeMap: {
              content: "tooltip",
            }
          }
        }
      }
    )
  )
)

test(
  "should add an extension with a mapped argument",
  t => t.snapshot(
    transformToHtml(
      "!icon(my-icon)",
      {
        elements: {
          icon: {
            attributeMap: {
              argument: "icon",
            }
          }
        }
      }
    )
  )
)

test(
  "should work with other markdown",
  t => t.snapshot(
    transformToHtml(
      `#heading

!icon(my-icon)`
    )
  )
)
