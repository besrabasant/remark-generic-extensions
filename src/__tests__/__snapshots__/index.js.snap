// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should add an extension 1`] = `
"<p><icon></icon></p>
"
`;

exports[`should add an extension with a class 1`] = `
"<p><icon class=\\"my-class\\"></icon></p>
"
`;

exports[`should add an extension with a lone attribute 1`] = `
"<p><icon attr=\\"\\"></icon></p>
"
`;

exports[`should add an extension with a lone attribute with a default value 1`] = `
"<p><icon attr=\\"default\\"></icon></p>
"
`;

exports[`should add an extension with a mapped argument 1`] = `
"<p><icon icon=\\"my-icon\\"></icon></p>
"
`;

exports[`should add an extension with a mapped content 1`] = `
"<p><icon tooltip=\\"my-tooltip\\"></icon></p>
"
`;

exports[`should add an extension with a quoted attribute 1`] = `
"<p><icon attr=\\"my attribute\\"></icon></p>
"
`;

exports[`should add an extension with an attribute 1`] = `
"<p><icon attr=\\"my-attr\\"></icon></p>
"
`;

exports[`should add an extension with an id 1`] = `
"<p><icon id=\\"my-id\\"></icon></p>
"
`;

exports[`should work with other markdown 1`] = `
"<p>#heading</p>
<p><icon></icon></p>
"
`;
